tests/references/test_oas.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from pydantic import ValidationError
+
+from virtool.references.oas import CreateReferenceSchema
+
+
+@pytest.fixture
+def mock_create_reference_schema():
+    return {
+        "name": "Test Viruses",
+        "description": "A bunch of viruses used for testing",
+        "data_type": "barcode",
+        "organism": "virus",
+    }
+
+
+@pytest.mark.parametrize(
+    "error", [None, "data_type", "remote_from", "two_values", "all_values"]
+)
+def test_schema(error, mock_create_reference_schema):
+    CreateReferenceSchema(**mock_create_reference_schema)
+
+    if error == "data_type":
+        with pytest.raises(ValidationError):
+            mock_create_reference_schema.update({"data_type": "bad type"})
```

Comment:
Put the update dictionary as the pytest parameter instead. This test can be heavily simplified.